libs/aws/langchain_aws/runnables/q_business.py

```diff
@@ -0,0 +1,162 @@
+                    'parentMessageId': self.parent_message_id,
+                })
+
+            # Call Amazon Q
+            response = self.client.chat_sync(**request)
+            self._last_response = response
+
+            # Extract the response text
+            if 'systemMessage' in response:
+                return response["systemMessage"]
+            else:
+                raise ValueError("Unexpected response format from Amazon Q")
+
+        except Exception as e:
+            if "Prompt Length" in str(e):
+                logging.info(f"Prompt Length: {len(input)}")
+                print(f"""Prompt:
+                {input}""")
+            raise ValueError(f"Error raised by Amazon Q service: {e}")
+
+    def get_last_response(self) -> Dict:
+        """Method to access the full response from the last call"""
+        return self._last_response
+
+    def validate_environment(self) -> Self:
+        """Don't do anything if client provided externally"""
+        #If the client is not provided, and the user_id is not provided in the class constructor, throw an error saying one or the other needs to be provided
+        if self.credentials is None:
+            raise ValueError(
+                "Either the credentials or the client needs to be provided."
+            )
+
+        """Validate that AWS credentials to and python package exists in environment."""
+        try:
+            import boto3
+
+            try:
+                if self.region_name is not None:
+                    client = boto3.client('qbusiness', self.region_name, **self.credentials)
+                else:
+                    # use default region
+                    client = boto3.client('qbusiness', **self.credentials)
+
+            except Exception as e:
+                raise ValueError(
+                    "Could not load credentials to authenticate with AWS client. "
+                    "Please check that credentials in the specified "
+                    "profile name are valid."
+                ) from e
+
+        except ImportError:
+            raise ImportError(
+                "Could not import boto3 python package. "
+                "Please install it with `pip install boto3`."
+            )
+        return client
+    def convert_langchain_messages_to_q_input(self, input: Any) -> str:
+        # Messages must be of type human', 'user', 'ai', 'assistant', or 'system
+        # Instead of logically formulating a message. We will allow langchain users to have their messages
+        # Added line by line the way they ordered them in the chain. We will prefix the content with the type,
+        # Hopefully this will inform Amazon Q how each message in the chain should be interpreted
+        messagesToStringArray = []
+        for message in input.to_messages(): # Returns List[BaseMessage]
+            messagesToStringArray.append(message.type + ": " + message.content)
+        return "\n".join(messagesToStringArray)
```

Comment:
PromptValue has a `to_string()` method that could possibly replace this entire block: https://python.langchain.com/api_reference/core/prompt_values/langchain_core.prompt_values.PromptValue.html#langchain_core.prompt_values.PromptValue

But test it out and ensure that the value returned is still returned in the desired format.